modules/dataframe/src/datastructures/dataframe.cpp

```diff
@@ -155,11 +155,13 @@ const std::vector<std::pair<std::string, const DataFormatBase *>> DataFrame::get
     return headers;
 }
 
-std::string DataFrame::getHeader(size_t idx) const { return columns_[idx]->getHeader(); }
+std::string DataFrame::getHeader(size_t idx) const { return columns_.at(idx)->getHeader(); }
```

Comment:
don't use exception in place of an assertion!

Reply:
Assertions are only enabled in debug mode....

Reply:
or when we force enable them in cmake

Reply:
that is a different issue
You should not depend on exception for detecting programmer errors!